packages/vaadin-menu-bar/src/vaadin-menu-bar-buttons-mixin.js

```diff
@@ -151,59 +166,100 @@ export const ButtonsMixin = (superClass) =>
       this.__renderButtons(this.items);
     }
 
-    /** @private */
-    __renderButtons(items = []) {
+    /** @protected */
+    _removeButtons() {
       const container = this._container;
-      const overflow = this._overflow;
 
       while (container.children.length > 1) {
         container.removeChild(container.firstElementChild);
       }
+    }
+
+    /** @protected */
+    _initButton(item) {
+      const button = document.createElement('vaadin-menu-bar-button');
+      button.setAttribute('part', 'menu-bar-button');
+
+      const itemCopy = Object.assign({}, item);
+      button.item = itemCopy;
+
+      if (item.component) {
+        const component = this.__getComponent(itemCopy);
+        itemCopy.component = component;
+        // save item for overflow menu
+        component.item = itemCopy;
+        button.appendChild(component);
+      } else if (item.text) {
+        button.textContent = item.text;
+      }
+
+      if (this.theme && this.theme !== '') {
+        button.setAttribute('theme', this.theme);
+      }
+
+      return button;
+    }
+
+    /** @protected */
+    _setButtonDisabled(button, disabled) {
+      if (disabled) {
+        button.disabled = true;
+        button.setAttribute('tabindex', '-1');
+      } else {
+        button.setAttribute('tabindex', '0');
+      }
+    }
+
+    /** @protected */
+    _setButtonAttrs(button) {
```

Comment:
It appears that the method only makes sense for just created buttons as old ones can be already expanded by the time calling this method.

Considering this, I wonder if it would be better to name the method like `_setInitialButtonAttrs` or `_initButtonAttrs`?

Reply:
Done.